utils/uri/matcher.go

```diff
@@ -0,0 +1,75 @@
+package uri
```

Comment:
Could you move this file to the `authorization/v2` package?

Also the URI struct should probably have its own separate file rather than including it in the same file as the logic for the matcher.

Reply:
Sure